Ceen.Httpd/BufferedStreamReader.cs

```diff
@@ -114,7 +117,7 @@ public async Task ReadHeaders(int maxlinesize, int maxheadersize, TimeSpan idlet
 						throw new HttpException(HttpStatusCode.RequestTimeout); 
 				}
 
-				var r = await rtask;
+				var r = rtask.Result;

```

Comment:
Is there a reason for not using await here? 

I find it "cleaner" to not access the `Result` property, as a "just in case it has not finished" guard.

At this point we know that `rtask` has finished so there is no delay imposed with using `await`, and either way works. But I was just wondering if there was a reason you prefer this way?

Reply:
Await will produce another state machine branch, which is unnecessary here.
And as we know that task completed, we can just take result from it.